### PR TITLE
Update to go 1.22

### PR DESCRIPTION
**Pull request by Luap99**
#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake

/kind other

#### What this PR does / why we need it:

Update golang so we can keep updating dependecies

#### Does this PR introduce a user-facing change?

```release-note
Buildah now requires go 1.22 to build.
```

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

File: go.mod [hunk 1]
```diff
@@ -1,21 +1,8 @@
 module github.com/containers/buildah
 
-// Minimum required golang version
-go 1.21.0 // *****  ATTENTION  WARNING  CAUTION  DANGER  ******
+// Warning: Ensure the "go" and "toolchain" versions match exactly to prevent unwanted auto-updates
```
Comment (edsantiago) [inline go.mod:3]: Will it be obvious to future maintainers what "toolchain" means? The string "toolchain" does not appear anywhere else in this file. My guess was, this means `test/tools/go.mod`, but that file has go 1.21

**Review by edsantiago — APPROVED** [review]
LGTM with one question

File: go.mod (hunk 1 above)
Comment (Luap99) [reply go.mod:3]: It is the same comment we have in podman and c/common and yes a go dev should now what these mean I would say, https://go.dev/doc/modules/gomod-ref#toolchain
Though I am happy to alter this comment to something else that makes it more clear, generally speaking we never want toolchain to be newer than the go version and it seems at some point the go vendor process decided to not add toolchain if it matches the exact go version, i.e. if we try to add `toolchain go1.22.0` make vendor removes it again. Only if you have something like `toolchain go1.22.1` it stays but this is not something we want.

So I guess we should simplify this comment and say we never ever want a toolchain directive in this file.

Comment (Luap99) [reply go.mod:3]: Can you merge this for now? 
I have some outstanding discussions around the handling of patch versions in the go line with @mtrmac so I like to wait for whatever we decide on and then update the comments based on that

**Comment by edsantiago** [comment]
Looks like I don't have approve privs in this repo

**Pull request merged**